Revit_Core_Engine/Create/Sheet.cs

```diff
@@ -0,0 +1,79 @@
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using Autodesk.Revit.DB;
+using BH.oM.Base.Attributes;
+using System.ComponentModel;
+
+namespace BH.Revit.Engine.Core
+{
+    public static partial class Create
+    {
+        /***************************************************/
+        /****               Public Methods              ****/
+        /***************************************************/
+
+        [Description("Creates and returns a new Floor Plan view in the current Revit file.")]
+        [Input("document", "Revit current document to be processed.")]
+        [Input("sheetName", "Name of the new sheet.")]
+        [Input("sheetNumber", "Number of the new sheet.")]
+        [Input("viewTemplateId", "The Title Block Id to be applied in the sheet.")]
+        [Output("viewSheet", "The new sheet.")]
+        public static ViewSheet Sheet(this Document document, string sheetName, string sheetNumber, ElementId titleBlockId)
+        {
+            ViewSheet result = null;
+
+            result = ViewSheet.Create(document, titleBlockId);
+
+            if (!string.IsNullOrEmpty(sheetName))
+            {
+                try
+                {
+                    result.Name = sheetName;
+                }
+                catch
+                {
+                    BH.Engine.Base.Compute.RecordWarning("There is already a sheet named '" + sheetName + "'." + " It has been named '" + result.Name + "' instead.");
+                }
+            }
+
+            if (!string.IsNullOrEmpty(sheetNumber))
+            {
+                try
+                {
+                    result.SheetNumber = sheetNumber;
+                }
+                catch
+                {
+                    BH.Engine.Base.Compute.RecordWarning("There is already a sheet with number '" + sheetNumber + "'." + " It has been named '" + result.SheetNumber + "' instead.");
```

Comment:
Instead of using the new number Revit auto-generates, should we append (1)/(2)/(3)/etc. to the number the user has specified? 
If so, a while loop to increment this number until the new name is unique will do. The user can then renumber the sheet as required much quicker.

Reply:
Good idea, will do it 👍